app/controllers/admin/enterprises_controller.rb

```diff
@@ -148,8 +150,15 @@ def find_resource
 
     private
 
-    def load_enterprise_set
-      @enterprise_set = Sets::EnterpriseSet.new(collection) if spree_current_user.admin?
+    def load_enterprise_set_on_index
+      return unless spree_current_user.admin?
+
+      load_enterprise_set_with_params
+    end
+
+    def load_enterprise_set_with_params(params = {})
+      @pagy, paginated_collection = pagy(@collection)
```

Comment:
I'm curious, why the extra local variable instead of just using the instance variable?

Reply:
You mean, something like:

```ruby
@pagy, @paginated_collection = pagy(@collection)
```
?

The only reason (which is not actually) I have is that it's done like this in the other part of the code.
But you're right, [documentation](https://github.com/ddnexus/pagy) is clear about this:
```ruby
# Wrap your collections with pagy in your actions
@pagy, @records = pagy(Product.all)
```

Conclusion: done! (and force pushed)

Reply:
Apologies for the confusion, I actually didn't understand the code and assumed that the same object was being stored in both variables. 
But I guess the result of the `pagy` method is an array with two elements, assigned to each variable. 

Good idea to use the convention from the docs 👍